W-22352449 *new* MuleSoft Platform MCP Server docs (#48)
mulesoft/docs-mulesoft-mcp-server: modules/ROOT/pages/getting-started-platform.adoc

= Getting Started with MuleSoft Platform MCP Server

> MuleSoft Platform MCP Server provides governed access to your entire IT landscape, including APIs, agents, MCP servers, and LLMs, to accelerate production-ready AI at scale. It helps you scan, discover, govern, and monitor services across the infrastructure you already run.

Reviewer: Verify if "MuleSoft Platform" is sanctioned.

Reviewer: Also, suggestion for the short description (to lead with user benefit rather than repeating the product name):
Scan, discover, govern, and monitor APIs, agents, MCP servers, and LLMs across your entire IT landscape from a single AI-accessible interface. MuleSoft Platform MCP Server gives AI tools governed access to your infrastructure to accelerate production-ready AI deployment. Connect your IDE in minutes using any MCP-compatible client.

Author: Love it!

== Create a Connected App

Follow the steps in xref:access-management::creating-connected-apps-dev.adoc#create-connected-app[Create a Connected App That Acts on the User’s Behalf] to create a connected app.

Additionally, when prompted, add the following permission scopes. These scopes enable you to use specific MuleSoft MCP Server tools. Make sure that all business groups and relevant environments are selected for these scopes.

> Refer to the xref:reference-mcp-tools-platform.adoc[MuleSoft Platform MCP Server Tool Reference] for details on all MuleSoft Platform MCP Server tools. Only tools listed in these tables require specific permission scopes.

Reviewer: ```suggestion
See the xref:reference-mcp-tools-platform.adoc[MuleSoft Platform MCP Server Tool Reference] for details on all MuleSoft Platform MCP Server tools. Only tools listed in these tables require specific permission scopes.
```

* xref:reference-mcp-tools-platform.adoc#test-scanner-connection[test_scanner_connection]
|===

=== General
[cols="2,1,3"]
|===
|To... |Required Permission |Associated Tools

|View organization information.
|organization_read (implicit) a|
* xref:reference-mcp-tools-platform.adoc#get-business-group[get_business_group]
* xref:reference-mcp-tools-platform.adoc#list-business-groups[list_business_groups]
* xref:reference-mcp-tools-platform.adoc#select-active-business-group[select_active_business_group]
* xref:reference-mcp-tools-platform.adoc#select-active-environment[select_active_environment]
* xref:reference-mcp-tools-platform.adoc#select-active-provider[select_active_provider]
* xref:reference-mcp-tools-platform.adoc#select-api-version[select_api_version]
|===

== Configure Your IDE

Configure Claude Desktop, Cursor, Windsurf, Zed, or other IDEs to work with MuleSoft Platform MCP Server. After you add the configuration, verify it by checking that the server has started in the MCP section of your IDE.

=== Available Client Connections

When configuring your AI tool, use the endpoint for your region.

> Use `/mcp`, and fall back to `/mcp/sse` if the client does not connect.

Reviewer: This isn't rendering correctly 146 - 149. I think because of the + sign.

+
* https://omni.mulesoft.com/mcp
* https://omni.mulesoft.com/mcp/sse

[cols="1,3"]
|===
|Type |URL
|US Prod
|https://omni.mulesoft.com[omni.mulesoft.com] (alias) / https://us1.omni.mulesoft.com[us1.omni.mulesoft.com]
|EU Prod
|https://eu1.omni.mulesoft.com[eu1.omni.mulesoft.com]
|CA Prod
|https://ca1.omni.mulesoft.com[ca1.omni.mulesoft.com]
|JP Prod
|https://jp1.omni.mulesoft.com[jp1.omni.mulesoft.com]
|IN Prod
|https://in1.omni.mulesoft.com[in1.omni.mulesoft.com]
|AU Prod
|https://au1.omni.mulesoft.com[au1.omni.mulesoft.com]
|AE Prod
|https://ae1.omni.mulesoft.com[ae1.omni.mulesoft.com]
|===

For more information about Anypoint Platform regions, see xref:hosting-home::index.adoc#control-plane-hosting-options[Control Plane Hosting Options].

=== Claude Desktop

In Claude Desktop:

1. Click **Customize** > **Connectors**.
2. Click the add (**+**) icon, and then click **Add Custom Connector**.
3. Add a name for the connection, and in the **Remote Universal URL** field, add the appropriate URL, for example `https://omni.mulesoft.com/mcp`.

> OR run this command in your terminal:

Reviewer: ```suggestion
Or run this command in your terminal:
```

|EU Prod
|https://eu1.omni.mulesoft.com[eu1.omni.mulesoft.com]
|CA Prod
|https://ca1.omni.mulesoft.com[ca1.omni.mulesoft.com]
|JP Prod
|https://jp1.omni.mulesoft.com[jp1.omni.mulesoft.com]
|IN Prod
|https://in1.omni.mulesoft.com[in1.omni.mulesoft.com]
|AU Prod
|https://au1.omni.mulesoft.com[au1.omni.mulesoft.com]
|AE Prod
|https://ae1.omni.mulesoft.com[ae1.omni.mulesoft.com]
|===

For more information about Anypoint Platform regions, see xref:hosting-home::index.adoc#control-plane-hosting-options[Control Plane Hosting Options].

=== Claude Desktop

In Claude Desktop:

. Click *Customize* > *Connectors*.
. Click the add (+) icon, and then click *Add Custom Connector*.
. Name the connection, and enter the *Remote Universal URL*, for example, https://omni.mulesoft.com/mcp[https://omni.mulesoft.com/mcp].
. Enter the Client ID and Client Secret from your connected app.

In your connected app:

. Log into Anypoint Platform, and open your connected app.
. In the *Redirect URLs* field, enter `https://claude.ai/api/mcp/auth_callback`.
. To apply changes, click *Continue*.

OR run this command in your terminal:

[source,bash]
----

> MCP_CLIENT_SECRET=<YOUR CLIENT SECRET> claude mcp add --transport http --client-id <YOUR CLIENT ID> --client-secret mulesoft-platform <URL for your region>

Reviewer: Cursor flagged this: The argument order looks off. mulesoft-platform appears to be the server name but is placed where <URL> should be, and <URL for your region> trails at the end without a flag. This could confuse users.

----

Download https://claude.com/download[Claude Desktop].

=== ChatGPT

In ChatGPT:

. Click *Apps* > *Settings.*
. Click the settings icon, click *Create App*, and configure the following:
.. Name the connection, and enter the *Remote Universal URL*, for example, https://omni.mulesoft.com/mcp[https://omni.mulesoft.com/mcp].
. In *OAuth* > *Advanced settings* > *Client registration*, configure the following:
.. Enter the Client ID and Client Secret from your connected app.
.. Copy the Callback URL.

In your connected app:

. Log in to Anypoint Platform, and open your connected app.
. In the redirect URLs field, enter the Callback URL value you copied.
. To apply changes, click *Continue*.

Download https://chatgpt.com/download/[ChatGPT].

=== Cursor

In Cursor:

. Click *Settings* > *Tools and MCPs* > *Add a Custom MCP Server*.
. Add the following code snippet:
+
[source,json]
----
{
  "mcpServers": {
    "mulesoft-platform": {
      "command": "npx",
      "args": [
        "-y",

>         "mcp-remote@latest",

Reviewer: This uses @latest

        "https://omni.mulesoft.com/mcp",
        "--static-oauth-client-info",
        "{\"client_id\":\"YOUR_CLIENT_ID\",\"client_secret\":\"YOUR_CLIENT_SECRET HERE\"}"
      ]
    }
  }
}
----

Download https://cursor.com/[Cursor].

=== Windsurf

In Windsurf:

. Click *Settings* > *Cascade* > *Open MCP Registry*.
. Click the settings icon and in `~/.codeium/windsurf/mcp_config.json` add the following code snippet:
+
[source,json]
----
{
  "mcpServers": {
    "mulesoft-platform": {
      "command": "npx",
      "args": [
        "-y",

>         "mcp-remote@0.1.38",

Reviewer: This is using a specific version (1.38) rather than latest. Might be intentional.

        "https://omni.mulesoft.com/mcp",
        "--static-oauth-client-info",
        "{\"client_id\":\"YOUR_CLIENT_ID\",\"client_secret\":\"YOUR_CLIENT_SECRET\"}"
      ]
    }
  }
}
----

Download https://windsurf.com/[Windsurf].

=== VS Code

In VS Code:

. In your user profile, configure a custom MCP server in the `mcp.json` file. Alternatively, run `MCP: Add Server` in the Command Palette (Shift+Cmd+P) to add a server through a guided flow.
+
[source,json]
----
{
  "servers": {
    "mulesoft-platform": {
      "type": "stdio",
      "command": "npx",
      "args": [
        "-y",

>         "mcp-remote@0latest",

Reviewer 2: ```suggestion
        "mcp-remote@latest",
```

For more information about Anypoint Platform regions, see xref:hosting-home::index.adoc#control-plane-hosting-options[Control Plane Hosting Options].

=== Claude Desktop

In Claude Desktop:

. Click *Customize* > *Connectors*.
. Click the add (+) icon, and then click *Add Custom Connector*.

> . Name the connection, and enter the *Remote Universal URL*, for example, https://omni.mulesoft.com/mcp[https://omni.mulesoft.com/mcp].

Reviewer: Is that a valid URL?
